harry-core/src/harry/data/ResultSetRow.java

```diff
@@ -55,12 +78,26 @@ public ResultSetRow clone()
                                 Arrays.copyOf(vds, vds.length), Arrays.copyOf(lts, lts.length));
     }
 
+    public boolean equals(Object o)
+    {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        ResultSetRow that = (ResultSetRow) o;
+        return pd == that.pd &&
+               cd == that.cd &&
+               Arrays.equals(vds, that.vds) &&
+               Arrays.equals(lts, that.lts) &&
+               Arrays.equals(sds, that.sds) &&
+               Arrays.equals(slts, that.slts) &&
+               Objects.equals(visited_lts, that.visited_lts);
+    }
+
     public String toString()
```

Comment:
nit: Does `visited_lts` need to be part of the `toString()`, or is it something that might grow to be potentially huge and not visually friendly? :D

Reply:
yeah, here i would probably leave it out since we display it separately